### ISSUE TITLE
Servermap is not showing all inter-connected applications

### ISSUE THREAD

**Issue by PIYUSHRAJ24** [pinpoint-apm/pinpoint#9841]
Hi,
I have configured pinpoint-apm agent in my **production servers** for Java applications but in servermap some components are showing but I am not able to see all the other inter-connected applications like **MongoDB** and others. Please suggest to me if there are any configurations that need to do or if some arguments need to pass while configuring the agent.  Please provide me a
 solution it would be very helpful.

Thanks
![Screenshot from 2023-04-11 15-02-07](https://user-images.githubusercontent.com/96586938/231122885-488e5ac1-87dd-4048-bb37-2ced3ccb03ea.png)

**Comment by PIYUSHRAJ24** [comment]
I have checked the file which you suggested above it was already configured by default and profiler settings also passed with recommended parameter but still it's now working. Previously I was using NewRelic it was working there but it's not working here.
Please guide me and provide the solution. It would be great.

Thankyou

**Comment by intr3p1d** [comment]
There could be several reasons, but the first reason your application is showing up as `Unknown` is because the metadata about the agent's serviceType is not being stored correctly.

If the logs on the agent side look fine (i.e., the agent is sending all the Span data correctly), it's worth checking if this information is being stored correctly on the collector side.

The information related to this metadata is usually stored in the serviceType(short) of the SpanBo. The information related to this metadata is usually stored in the serviceType of SpanBo. https://github.com/pinpoint-apm/pinpoint/blob/796bd8d239cac27430550f649a78644196b3d398/plugins/mongodb/src/main/java/com/navercorp/pinpoint/plugin/mongo/MongoConstants.java#L85-L87 If you follow this link, you can see the constants that tell the API calling MongoDB that it is MongoDB. It's a good idea to check that these values are passed to the serviceType properly. If they are not recognized, they are probably being stored as `Unknown`.
https://github.com/pinpoint-apm/pinpoint/blob/046d24f0ec6e731f1ff85c26b562353a40eec0b1/commons/src/main/java/com/navercorp/pinpoint/common/trace/ServiceType.java#L327-L336
You can also check if MongoDB-related APIs appear as unknown in the transaction view when dragged from the scatter chart on the top right.

If the values are being stored correctly in the collector, but are showing up in the unknown group, it's a web issue. If they are not being saved, it is recommended to find out why they are not being saved in the collector. The collector's log for analysis is usually saved as `/logs/pinpoint-collector-yyyy-mm-dd-n.log`.

To summarize
1. if the SpanBo's ServiceType itself is passed as `Unknown` when passed to the collector, this is because the agent does not properly recognize the MongoDb you are using. This may be due to out-of-scope support for MongoDB.
2. if your SpanBo has the correct ServiceType and you don't have any issues when saving to the DB, it could be a web issue.
3. or there may be a problem storing the SpanBo in the collector.

**Comment by PIYUSHRAJ24** [comment]
Basically, my MongoDb clusters are running on a different server, and the Java application is running on different servers. Now my concern is that I have installed the pinpoint agent in the Java application server and how could it monitor MongoDB?

**Comment by PIYUSHRAJ24** [comment]
Please update

**Comment by intr3p1d** [comment]
> how could it monitor MongoDB?

It monitor MongoDB via used APIs. Maybe it would be out-of-scope support.
Pinpoint only supports Mongo Client, so there's nothing else you need to install on your Mongo cluster.

**Comment by PIYUSHRAJ24** [comment]
Do I need to install the agent in the MongoDB cluster and need to enable profiling?

**Comment by PIYUSHRAJ24** [comment]
Please update
